docs_devel/docs/12.HowToMakeMTConnector.md

```diff
@@ -0,0 +1,112 @@
+- [Autshumato MT](https://sourceforge.net/projects/autshumatoite/files/OmegaT-plugins-AutshumatoMT/)
+- [Mirai translator](https://codeberg.org/miurahr/omegat-mirai/releases)
+- [NICT TexTra](https://codeberg.org/miurahr/omegat-textra-plugin/releases)
+- [Tencent translation](https://github.com/yoyicue/omegat-tencent-plugin)
+
+## Getting started the project
+
+OmegaT project provides a plugin project skeleton repository. You can start your project from
+a skeleton project. Please go to the URL https://github.com/omegat-org/plugin-skeleton and
+click "Use this template" button to start your project in GitHub.
+When you use codeberg.org forge site, there is also a skeleton at https://codeberg.org/miurahr/omegat-plugin-skeleton 
+
+The skeleton project uses Gradle for a build system.
+You can select Groovy DSL or Kotlin DSL for the configuration.
+When you choose a Groovy DSL, please rename `build.gradle.disabled` to `build.gradle` and remove `build.gradle.kts`.
+
+You should also modify a file
+- Project.name in settings.gradle
+- Properties: description, title, website and category
+- Plugin Main class name in build.gradle(.kts).
+
+Implementation should be placed at
+- Source code: src/main/<lang>/*
+- Test code: src/test/<lang>/* and src/test/resources/*
+
+## OmegaT MT API - `org.omegat.gui.exttrans.IMachineTranslation`
+
+### String getTranslation(Language sLang, Language tLang, String text) throws Exception;
+
+It is a main method to override in your plugin project. The method recieves a source text, translation from sLang to 
+tLang language, and return translation text.
+
+### String getCachedTranslation(Language sLang, Language tLang, String text);
+
+It is another method that you should override. If your plugin has a caching mechanism, and there is entry for the text,
+return translation without accessing MT engine. Otherwise, return null.
+
+> **_NOTE:_**  You should not call the engine in the method, because OmegaT will call it even when the target 
+> segment has already translated, but configured as `MT only untranslated segment` is on or `automatically translate` is off.
+
+### getName()
+
+Human and machine-readable name of MT engine. OmegaT uses it for the index of results, and as a property value, and 
+logging purpose.
+
+### isEnabled()/setEnabled(boolean b)
+
+return and set an enabled status.
+
+### boolean isConfigurable() /  void showConfigurationUI(Window parent)
+
+When the plugin return true for the method, `isConfigurable()` OmegaT will call `showConfigurationUI` when user 
+clicks a configuration UI button on `Tools > Preferences > MachineTranslation`.
+
+## Convenience abstract class
+
+There are two convenience class for plugins. You are recommended to use `org.omegat.core.
+machinetranslators.BaseCachedTranslate`. It implements many necessary things, and you can concentrate into a logic to 
+access your MT engine API.  There are only three methods you should override.
```

Comment:
We should express about how to handle credentials to access API. Also we should describe how to implement a preferences dialog to ask credential information and how to store the credentials in a persistent storage.